rfcs/006-token-based-extension-system.md

```diff
@@ -0,0 +1,44 @@
+# Token-based Extension System for Assessments
+
+## Introduction
+
+The purpose of this RFC is to propose a token-based extension system for assessments in order to provide students with a flexible mechanism for requesting deadline extensions. This system will allow students to utilize tokens they receive at the beginning of the semester to apply for extensions on their assessments. Instructors will have control over the allocation and usage of tokens within defined assessment sets, allowing them to manage and customize extension policies according to their course requirements.
+
+## Token Allocation
+
+2.1 Initial Allocation: At the start of the semester, all students will receive a fixed number of tokens. The exact number of tokens per student should be determined based on factors such as the course workload and assessment distribution.
+
+2.2 Additional Tokens: Instructors will have the discretion to assign additional tokens to individual students on a case-by-case basis if deemed necessary. This flexibility allows instructors to accommodate special circumstances or provide extra support to specific students.
+
+## Token Usage
+
+3.1 Extension Types: Each token will represent a specific type of extension, and the following options are proposed as initial suggestions:
+a) One token = one day extension
+b) One token = one week extension
+c) One token = student selects the new due date
+
+3.2 Token Constraints: Instructors will define the assessment sets where tokens can be utilized at the course instance level. Assessment sets can include various types of assessments such as homework, quizzes, exams, machine problems, etc. By specifying token constraints, instructors can control which assessments are eligible for extension requests using tokens. This provides instructors with the flexibility to tailor the extension policy to suit the nature and requirements of each assessment.
+
+## Requesting Extensions
+
+4.1 Student Request: When a student requires an extension on a specific assessment within an eligible assessment set, they can submit a request indicating the desired length of the extension.
+
+4.2 Token Consumption: Upon approval of the extension request, the system will deduct the appropriate number of tokens from the student's token balance based on the chosen extension type.
+
+4.3 Token Validation: The system will verify if the student has a sufficient number of tokens to cover the requested extension. If the student lacks the necessary tokens, the request will be automatically rejected.
+
+## Considerations and Future Enhancements
```

Comment:
Is it fair to say that these are all outside the scope of an initial implementation of this?

Reply:
Yes, these are future considerations and enhancements. It won't be a part of the initial implementation.